realm/realm-library/src/main/java/io/realm/StandardRealmObjectSchema.java

```diff
@@ -676,6 +687,12 @@ private void checkFieldExists(String fieldName) {
         }
     }
 
+    private void checkAddPrimaryKeyForSync() {
```

Comment:
Since migrations work very differently across bindings, I'm not sure that is needed?